lib/kafka/async_producer.rb

```diff
@@ -274,6 +282,7 @@ def produce(value, **kwargs)
             sleep @retry_backoff**retries
             retry
           else
+            # @finally(value)
```

Comment:
We should definitely get in here with the `@finally` callable so that we can make the number of retries finite.